libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -586,7 +586,7 @@ bool ExpressionCompiler::visit(FunctionCall const& _functionCall)
 			m_context << Instruction::CREATE;
 			// Check if zero (out of stack or not enough balance).
 			m_context << Instruction::DUP1 << Instruction::ISZERO;
-			m_context.appendConditionalInvalid();
+			m_context.appendConditionalRevert();
```

Comment:
I guess there are two types of inputs that might require validation. Obviously, there are method parameters that the caller provides. However, I would include return values of callee methods as well.

For instance, method A might call a method B and expect the return value to be non-zero (e.g., to prevent division-by-zero later in the code). Since method A cannot always control the return value of B it seems fine to revert in such cases.

In other words, it's not a bug in A if B returns 0. Does that make sense?